build.gradle

```diff
@@ -11,11 +14,28 @@ plugins {
     id 'io.github.gradle-nexus.publish-plugin' version '1.0.0'
     id 'com.github.ben-manes.versions' version '0.51.0'
     id "me.champeau.jmh" version "0.7.3"
+    id "net.ltgt.errorprone" version '4.2.0'
+
+    // Kotlin just for tests - not production code
+    id 'org.jetbrains.kotlin.jvm' version '2.1.21'
 }
 
 java {
     toolchain {
-        languageVersion = JavaLanguageVersion.of(11)
+        languageVersion = JavaLanguageVersion.of(17)
```

Comment:
needs to run on 17 to get errorprone happy (compiled for 17)

but we target 11 later